exchange/currency.go

```diff
@@ -0,0 +1,187 @@
+package exchange
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"strconv"
+)
+
+const defaultCurrencyConversionError = "No currency conversion available."
+
+type currencyData struct {
+	DataAsOfDate    string                      `json:"dataAsOf"`
+	ConversionRates openrtb_ext.ConversionRates `json:"conversions"`
+}
+
+func getCurrency(currencyRates []byte, seatBids *[]openrtb.SeatBid, bidRequest *openrtb.BidRequest, adapterExtra *map[openrtb_ext.BidderName]*seatResponseExtra) string {
+	currencyToUse := defaultAdServerCurrency
+	requestCurrency := defaultAdServerCurrency // Assuming default USD for request currency if one was not set
+
+	if len(bidRequest.Cur) == 1 {
+		// Setting request currency if one was set
+		requestCurrency = bidRequest.Cur[0]
+	}
+
```

Comment:
Took me a while to see the point of `currencyToUse`. If `requestCurrency` is not `defaultAdServerCurrency`, but there were no valid bids (currency conversions failed), then `currencyToUse` will be `defaultAdServerCurrency`.